### DISCUSSION TITLE
Builder /Token Problem

### DISCUSSION THREAD

**Discussion by pierremartel** [lcobucci/jwt#889] [Q&A]
I try to implement a Real time Chat in my app on Symfony with Mercure but i have some problem with the builder . They give me this error:

[
![erreur_navigateur](https://user-images.githubusercontent.com/98807527/186933760-684f1a4d-d4c8-41eb-9e8f-642ddbf7b6e6.png)
](url)

Here my code : 
[
![back jpg](https://user-images.githubusercontent.com/98807527/186933665-fda370bb-9352-4404-be1b-ba67162700f8.png)
](url)
[
![](https://user-images.githubusercontent.com/98807527/186933897-c79e1ae4-8dd2-4fb7-83af-7631e2181c25.png)
](url)

So i try to change my code with the configuration instance because i think it's about my version of Lcobucci (v.4) and look at the documentation .

[
![update_1](https://user-images.githubusercontent.com/98807527/186935351-51b83dc0-c021-4543-8320-3df007f6ad8a.png)
](url)

[
![update_2](https://user-images.githubusercontent.com/98807527/186935401-653380e9-8f76-46f9-a896-125ce0370a6e.png)
](url)

And now i have this error : 

Cannot autowire service "App\Controller\IndexController": argument "$config" of method "__construct()" references class "Lcobucci\JWT\Configuration" but no such service exists.

What is wrong with my code ?

**Comment by Ocramius** [comment]
You are missing a service definition that leads to a `Lcobucci\JWT\Configuration` instance, as the exception states.

**Reply by pierremartel** [reply]
I try that at the line 26- 27 in my service.yaml but doesnt  work : 
[
![update_3](https://user-images.githubusercontent.com/98807527/186945816-c653a9d0-ecaa-40e5-b4ac-d4f4b092ba9e.png)
](url)

Is it right ? 

i have this error now  but i dont understand what to do : 

The controller for URI "/" is not callable: Controller "App\Controller\IndexController" cannot be fetched from the container because it is private. Did you forget to tag the service with "controller.service_arguments"?

**Reply by Ocramius** [reply]
No, that way you configured the `IndexController` to be an instance of `Lcobucci\JWT\Configuration`.

Please make sure that you **study** the [Symfony dependency injection documentation](https://symfony.com/doc/current/components/dependency_injection.html).

What you want is a n `Lcobucci\JWT\Configuration` service, configured in all its relevant details.
